bin/common.c

```diff
@@ -57,6 +57,74 @@ static uint8_t unsafe_verify_host_fn(const char *host_name, size_t host_name_len
     return 1;
 }
 
+int write_array_to_file(const char *path, uint8_t *data, size_t length) {
+    GUARD_EXIT_NULL(path);
+    GUARD_EXIT_NULL(data);
+
+    FILE *file = fopen(path, "wb");
+    if (!file) {
+       return S2N_FAILURE;
+    }
+
+    if(fwrite(data, sizeof(char), length, file) != length) {
+        fclose(file);
+        return S2N_FAILURE;
+    }
+    fclose(file);
+
+    return S2N_SUCCESS;
+}
+
+int get_file_size(const char* path, size_t *length)
+{
+    GUARD_EXIT_NULL(path);
+    GUARD_EXIT_NULL(length);
+
+    FILE *file = fopen(path, "rb");
+    if (!file) {
+       return S2N_FAILURE;
+    }
+
+    if(fseek(file, 0, SEEK_END) < 0) {
+        fclose(file);
+        return S2N_FAILURE;
+    }
+
+    long file_length = ftell(file);
+    if (file_length < 0) {
+        fclose(file);
+        return S2N_FAILURE;
+    }
+
+    *length = file_length;
+    fclose(file);
+    return S2N_SUCCESS;
+}
+
+int load_file_to_array(const char *path, uint8_t *data, size_t max_length)
```

Comment:
should this return `S2N_RESULT` instead and some of the checks do `RESULT_ENSURE_OK` where possible?

Reply:
Hmmmmm well this isn't an s2n function, its just a helper function I wrote. In order to use the s2n safety macros, we'd have to import them in, which I don't want to do, because s2nc is supposed to be how a customer would use s2n, and a customer shouldn't be using our safety macros.

Reply:
> customer shouldn't be using our safety macros.
might be out of scope for this PR but i wonder if its possible to suggest that as a good best practice for writing apps for our client. a subsset of them should be general